tests/test_integrated_channels/test_canvas/test_client.py

```diff
@@ -86,3 +89,144 @@ def test_client_instantiation_fails_without_refresh_token(self):
             canvas_api_client = CanvasAPIClient(self.enterprise_config)
             canvas_api_client._create_session()  # pylint: disable=protected-access
         assert client_error.value.__str__() == "Failed to generate oauth access token: Refresh token required."
+
+    def test_course_delete_fails_with_empty_data(self):
```

Comment:
hmm, are the empty data cases actual failures? I think generally the transmitter should not call with empty data but if it does, is it just a 'meh, no data' vs an error ?

Reply:
https://github.com/edx/edx-enterprise/pull/932#discussion_r476435593

> data is empty/None

The edge case handling here was in response to this comment, which for the most part I get the sense is impossible/incredibly unlikely to happen but I thought it was a relatively small error check and why not add it to be careful  `¯\_(ツ)_/¯`

Reply:
maybe an error is fine, just wondering if this empty data case can ever even occur .. does not hurt to add this protection though, just wondering about error vs pass ... error maybe slightly better since we will get the logging in this case at least ... just want to hear your thoughts, but no issues here, resolving this convo ...